server/channels/app/channel_category.go

```diff
@@ -52,16 +56,23 @@ func (a *App) GetSidebarCategoriesForTeamForUser(c request.CTX, userID, teamID s
 
 func (a *App) GetSidebarCategories(c request.CTX, userID string, opts *store.SidebarCategorySearchOpts) (*model.OrderedSidebarCategories, *model.AppError) {
 	var appErr *model.AppError
-	categories, err := a.Srv().Store().Channel().GetSidebarCategories(userID, opts)
-	if err == nil && len(categories.Categories) == 0 {
-		// A user must always have categories, so migration must not have happened yet, and we should run it ourselves
-		categories, appErr = a.createInitialSidebarCategories(userID, opts)
-		if appErr != nil {
-			return nil, appErr
-		}
+	appsCategoryEnabled := a.Config().FeatureFlags.AppsSidebarCategory
+	options := &store.SidebarCategorySearchOpts{
+		TeamID:              opts.TeamID,
+		ExcludeTeam:         opts.ExcludeTeam,
+		AppsCategoryEnabled: appsCategoryEnabled,
 	}
-
-	if err != nil {
+	categories, err := a.Srv().Store().Channel().GetSidebarCategories(userID, options)
+	if err == nil {
+		if len(categories.Categories) == 0 || (appsCategoryEnabled && checkMissingSystemSidebarCategories(categories)) {
+			// A user must always have system categories, so migration must not have happened yet, and we should run it ourselves
+			categories, appErr = a.createInitialSidebarCategories(userID, options)
```

Comment:
If this is an upgrade, they will have the apps category missing and will execute the `createInitialSidebarCategories` again - will this cause any duplicates in the other system categories that were already existent?